index.js

```diff
@@ -199,6 +199,70 @@ function session(options){
 
       setcookie(res, name, req.sessionID, secrets[0], cookie.data);
     });
+    
+    // proxy redirect() to commit the session
+    var _redirect = res.redirect;
+    var redirected = false;
+    res.redirect = function redirect(url) {
```

Comment:
This will break everyone in Express doing `res.redirect(301, url)`, which is a valid call.